# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/loader/ScriptLoadRequest.h
# Commit: d2647082c823
# Full Hash: d2647082c823595eddd3e051bd99c3cbfc53ae73
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-24 21:46:01
# Description:
#   Bug 1756550 - propagate webext global to all dynamically imported modules; r=jonco
#   
#   The crashes were occurring due to the missing webext global on dynamically imported modules from a referencing script. There is likely a more elegant way to reintroduce this, but for now this is a quick fix.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139511
# ==============================================================================

diff -r e076ff6e02d1 -r d2647082c823 js/loader/ScriptLoadRequest.h
--- a/js/loader/ScriptLoadRequest.h	Thu Feb 24 12:18:19 2022 +0000
+++ b/js/loader/ScriptLoadRequest.h	Thu Feb 24 12:25:25 2022 +0000
@@ -77,7 +77,8 @@
 
   ScriptFetchOptions(mozilla::CORSMode aCORSMode,
                      enum mozilla::dom::ReferrerPolicy aReferrerPolicy,
-                     nsIPrincipal* aTriggeringPrincipal);
+                     nsIPrincipal* aTriggeringPrincipal,
+                     nsIGlobalObject* aWebExtGlobal = nullptr);
 
   /*
    *  The credentials mode used for the initial fetch (for module scripts)
@@ -93,9 +94,15 @@
   const enum mozilla::dom::ReferrerPolicy mReferrerPolicy;
 
   /*
-   * related to cryptographic nonce, used to determine CSP
+   *  Related to cryptographic nonce, used to determine CSP
    */
   nsCOMPtr<nsIPrincipal> mTriggeringPrincipal;
+
+  /* The Web Extension global -- Only used on DOM Modules.
+   *     Specifies a SandBox global with which to associate and run this script.
+   *     Propagated throughout the module tree if present. Not part of a spec.
+   */
+  nsCOMPtr<nsIGlobalObject> mWebExtGlobal;
 };
 
 /*
